app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/DatabaseChangelog.java

```diff
@@ -1993,4 +1995,79 @@ public void changeAppLayoutTypeDefinition(MongoOperations mongoOperations, Mongo
         }
     }
 
+    @ChangeSet(order = "060", id = "update-mysql-postgres-mongo-ssl-mode", author = "")
+    public void updateMysqlPostgresMongoSslMode(MongoTemplate mongoTemplate) {
+        Plugin mysqlPlugin = mongoTemplate
+                .find(query(where("name").is("Mysql")), Plugin.class).get(0);
+
+        Plugin mongoPlugin = mongoTemplate
+                .find(query(where("name").is("MongoDB")), Plugin.class).get(0);
+
+        List<Datasource> mysqlAndMongoDatasources = mongoTemplate
+                .find(
+                        query(new Criteria()
+                                .orOperator(
+                                        where("pluginId").is(mysqlPlugin.getId()),
+                                        where("pluginId").is(mongoPlugin.getId())
+                                )
+                        ),
+                        Datasource.class);
+
+        /*
+         * - Set SSL mode to DEFAULT for all mysql and mongodb datasources.
+         */
+        mysqlAndMongoDatasources
+                .stream()
+                .forEach(datasource -> {
+                    if(datasource.getDatasourceConfiguration() != null) {
+                        if(datasource.getDatasourceConfiguration().getConnection() == null) {
+                            datasource.getDatasourceConfiguration().setConnection(new Connection());
+                        }
+
+                        if(datasource.getDatasourceConfiguration().getConnection().getSsl() == null) {
+                            datasource.getDatasourceConfiguration().getConnection().setSsl(new SSLDetails());
+                        }
+
+                        datasource.getDatasourceConfiguration().getConnection().getSsl().setAuthType(SSLDetails.AuthType.DEFAULT);
+                    }
+
+                    mongoTemplate.save(datasource);
```

Comment:
Save the datasource only when we're making changes to it. We don't need to otherwise, right?

```suggestion
                        mongoTemplate.save(datasource);
                    }
```